azext_edge/edge/providers/rpsaas/adr/user_strings.py

```diff
@@ -6,6 +6,7 @@
 
 # Asset Strings
 ENDPOINT_NOT_FOUND_WARNING = "Endpoint {0} not found. The asset may fail provisioning."
+INVALID_OBSERVABILITY_MODE_ERROR = "{0} has an invalid observability mode [{1}]."
```

Comment:
Should this error tell them which observability mode is invalid? It seems to be used to show the valid modes in the array `[{1}]`, but it doesn't show which one is wrong. Maybe something like : `{0} has invalid observability mode "{1}", valid modes are: [{2}]` is more descriptive?

Reply:
there is one observability mode per event or data point. So I used the event notifier/data source (aka what is required) to help the customer identify which one is invalid. 

I didn't think there is any need to echo the customer input for the wrong observability mode back but if you feel strongly about this, I can add it in.